crates/rustfix/src/lib.rs

```diff
@@ -270,3 +280,34 @@ pub fn apply_suggestions(code: &str, suggestions: &[Suggestion]) -> Result<Strin
     }
     fix.finish()
 }
+
+/// Applies multiple `suggestions` to the given `code`.
+///
+/// When a suggestion has multiple solutions that they are applied separately
+/// each output being the result of only applying the Nth solution. If a
+/// suggestion has fewer solutions than other suggestions the first solution
+/// is repeated.
+///
+/// Currently individual subdiagnostics with multiple solutions are not
+/// detected, they are treated as a single combined solution. See
+/// [rust-lang/rust#53934](https://github.com/rust-lang/rust/issues/53934).
+pub fn apply_suggestions_separately(
```

Comment:
Do we consider duplicate solutions inside a suggestion? We should only apply once in that situation. See <https://github.com/rust-lang/cargo/pull/13728>.
That said, it can be considered a rustc bug and we do nothing here if the only consumer is clippy.